opendevin/llm/llm.py

```diff
@@ -54,9 +64,32 @@ def __init__(self,
         self.api_key = api_key
         self.base_url = base_url
         self.api_version = api_version
+        self.max_input_tokens = max_input_tokens
+        self.max_output_tokens = max_output_tokens
+        self.custom_llm_provider = custom_llm_provider
+
+        # litellm actually uses base Exception here for unknown model
+        self.model_info = None
+        try:
+            self.model_info = litellm.get_model_info(self.model_name)
+        # noinspection PyBroadException
+        except Exception:
+            logger.warning(f'Could not get model info for {self.model_name}')
+
+        if self.max_input_tokens is None:
+            if self.model_info is not None and 'max_input_tokens' in self.model_info:
+                self.max_input_tokens = self.model_info['max_input_tokens']
+            else:
+                self.max_input_tokens = 4096
+
+        if self.max_output_tokens is None:
+            if self.model_info is not None and 'max_output_tokens' in self.model_info:
+                self.max_output_tokens = self.model_info['max_output_tokens']
+            else:
+                self.max_output_tokens = 1024
```

Comment:
Just curious: where does this number come from? I guess 4096 is because it's the limit of GPT 3.5, but how about this one?

Reply:
I don't have a significant justification for either of these defaults, and I am interested to hear opinions on them. I regularly experienced overruns with a 512 output token limit, and therefore I usually use 1024 or higher locally.

Reply:
I don't have a strong opinion either. I just feel like it would be better to have some comments explaining where these numbers are from.

Reply:
I have added comments documenting this:

# Max input tokens for gpt3.5, so this is a safe fallback for any potentially viable model
self.max_input_tokens = 4096

# Enough tokens for most output actions, and not too many for a bad llm to get carried away responding
# with thousands of unwanted tokens
self.max_output_tokens = 1024